x-pack/plugins/apm/server/routes/metrics/by_agent/serveless/concurrent_invocations.ts

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { ProcessorEvent } from '@kbn/observability-plugin/common';
+import { kqlQuery, rangeQuery } from '@kbn/observability-plugin/server';
+import { euiLightVars as theme } from '@kbn/ui-theme';
+import {
+  SERVICE_NAME,
+  SERVICE_NODE_NAME,
+} from '../../../../../common/elasticsearch_fieldnames';
+import { environmentQuery } from '../../../../../common/utils/environment_query';
+import { getVizColorForIndex } from '../../../../../common/viz_colors';
+import { getMetricsDateHistogramParams } from '../../../../lib/helpers/metrics';
+import { Setup } from '../../../../lib/helpers/setup_request';
+import { GenericMetricsChart } from '../../fetch_and_transform_metrics';
+
+export async function getConcurrentInvocations({
+  environment,
+  kuery,
+  setup,
+  serviceName,
+  start,
+  end,
+}: {
+  environment: string;
+  kuery: string;
+  setup: Setup;
+  serviceName: string;
+  start: number;
+  end: number;
+}): Promise<GenericMetricsChart> {
+  const { apmEventClient, config } = setup;
+
+  const params = {
+    apm: {
+      events: [ProcessorEvent.metric],
+    },
+    body: {
+      size: 0,
+      query: {
+        bool: {
+          filter: [
+            { term: { [SERVICE_NAME]: serviceName } },
+            ...rangeQuery(start, end),
+            ...environmentQuery(environment),
+            ...kqlQuery(kuery),
+          ],
+        },
+      },
+      aggs: {
+        concurrentInvocations: {
+          terms: {
+            field: SERVICE_NODE_NAME,
+            // TODO: check if size 10 is enough
+          },
```

Comment:
Uses a `terms` aggregation to count the number of concurrent invocations. @AlexanderWert the terms aggregation by default returns 10 items, is it enough? too many? or must it be configurable?

Reply:
Not sure, terms is what we need here. We are not interested in the actual `service.node_name` values but only in the number of unique values for that field. 

I think we need the `cardinality` aggregation here. When selecting the corresponding config in Lens this is what I get:

<img width="419" alt="image" src="https://user-images.githubusercontent.com/866830/185308400-cbb3a483-fdb2-45e6-b3fd-ffb5bd2a23da.png">

<img width="736" alt="image" src="https://user-images.githubusercontent.com/866830/185308192-e14bfb99-6310-47e3-97be-44d495e6840e.png">

Reply:
okay, I got it now. thanks